controller/src/main/java/org/jboss/as/controller/PrimitiveListAttributeDefinition.java

```diff
@@ -102,6 +104,38 @@ public void removeCapabilityRequirements(OperationContext context, Resource reso
         handleCapabilityRequirements(context, resource, attributeValue, true);
     }
 
+    @Override
+    ModelNode parseResolvedValue(ModelNode original, ModelNode resolved) {
+        return parseSingleElementToList(this, original, resolved);
+    }
+
+    static ModelNode parseSingleElementToList(AttributeDefinition ad, ModelNode original, ModelNode resolved) {
+        ModelNode result = resolved;
+        if (original.isDefined()
+                && !resolved.equals(original)
+                && resolved.getType() == ModelType.LIST
+                && resolved.asInt() == 1) {
```

Comment:
is this correct? this would only pass if value would be "1" while having type of LIST which is kind a hard.

did you maybe mean resolve.asList().size() == 1?

Reply:
asInt() for ModelType.LIST returns the list size. 

https://github.com/jbossas/jboss-dmr/blob/master/src/main/java/org/jboss/dmr/ListModelValue.java#L104

For OBJECT it returns the number of keys.